### PR TITLE
feat: path query proof construction and verification without subquery support

### PR DESCRIPTION
**Pull request by iammadab**
Given a path to a merk and a query to be applied to that merk, 
we should apply the query to the merk, then show the merk
is contained in it's parent, parent is in their own parent up to
the root.

Summary: prove the query then prove the path.

### CODE REVIEW

File: grovedb/src/lib.rs [hunk 1]
```diff
@@ -68,7 +64,7 @@ pub enum Error {
     CorruptedData(String),
 }
 
-#[derive(Debug)]
+#[derive(Debug, Clone)]
```
Comment (fominok) [inline grovedb/src/lib.rs:67]: It is necessary to be `Clone` ?

**Review by fominok — CHANGES_REQUESTED** [review]

**Review by fominok — APPROVED** [review]

File: grovedb/src/lib.rs (hunk 1 above)
Comment (iammadab) [reply grovedb/src/lib.rs:67]: Resolved: changes to be made is subsequent pr

File: grovedb/src/operations/proof.rs [hunk 2]
```diff
@@ -1,261 +1,202 @@
...
+
+        merk_optional_tx!(self.db, path_slices.clone(), None, subtree, {
+            // TODO: Not allowed to create proof for an empty tree (handle this)
+            let proof = subtree
+                .prove(query.query.query, query.query.limit, query.query.offset)
+                .expect("should generate proof");
+
+            // TODO: Switch to variable length encoding
+            debug_assert!(proof.len() < 256);
+            write_to_vec(&mut proof_result, &vec![MERK_PROOF, proof.len() as u8]);
```
[the remark this answers is not in the document]
Comment (iammadab) [reply grovedb/src/operations/proof.rs:35]: Resolved: changes to be made is subsequent pr

File: grovedb/src/operations/proof.rs [hunk 3]
```diff
@@ -1,261 +1,202 @@
...
+    dest.write_all(value);
+}
+
+impl GroveDb {
+    pub fn prove(&self, query: PathQuery) -> Result<Vec<u8>, Error> {
+        let mut proof_result: Vec<u8> = vec![];
+
+        let path_slices = query.path.iter().map(|x| x.as_slice()).collect::<Vec<_>>();
+
+        self.check_subtree_exists_path_not_found(path_slices.clone(), None, None)?;
```
[the remark this answers is not in the document]
Comment (iammadab) [reply grovedb/src/operations/proof.rs:25]: Resolved: Change postponed

File: grovedb/src/operations/proof.rs [hunk 4]
```diff
@@ -1,261 +1,202 @@
...
+
+fn write_to_vec<W: Write>(dest: &mut W, value: &Vec<u8>) {
+    dest.write_all(value);
+}
+
+impl GroveDb {
+    pub fn prove(&self, query: PathQuery) -> Result<Vec<u8>, Error> {
+        let mut proof_result: Vec<u8> = vec![];
+
+        let path_slices = query.path.iter().map(|x| x.as_slice()).collect::<Vec<_>>();
```
[the remark this answers is not in the document]
Comment (iammadab) [reply grovedb/src/operations/proof.rs:23]: Resolved: Change postponed

**Pull request merged**